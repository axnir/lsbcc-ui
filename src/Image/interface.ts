import type { ImgHTMLAttributes, ReactNode } from 'react';

export interface ImageProps extends ImgHTMLAttributes<HTMLImageElement> {
  /**
   * @description      图片地址
   */
  src: string;
  /**
   * @description      可选，加载过程占位元素
   */
  fallback?: string;
  /**
   * @description      可选，加载失败的兜底图片
   */
  loader?: ReactNode;
}
