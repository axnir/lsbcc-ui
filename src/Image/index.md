---
nav:
  title: 组件
  path: /components
  order: 2
group:
  title: 组件
---

# Image 图片

展示图片。

## 代码演示

### 基本用法

<code src="./demo/index.tsx"></code>

<API src="./image.tsx"></API>
