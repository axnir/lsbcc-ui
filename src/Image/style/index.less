@imagePrefix: lsbcc-image;

.@{imagePrefix} {
  position: relative;

  .img-skeletion {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    opacity: .6;
    background: linear-gradient(90deg, #f2f2f2 25%, #e6e6e6 37%, #f2f2f2 63%);
    background-size: 400% 100%;
    animation: imgSkeleton .8s ease infinite;
  }
}

@keyframes imgSkeleton {
  0% {
    background-position: 100% 50%;
  }

  100% {
    background-position: 0 50%;
  }
}