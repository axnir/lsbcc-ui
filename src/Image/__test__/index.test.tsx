import React from 'react';
import { render } from '@testing-library/react';
import Image from '../image';

const src = 'https://s3.bmp.ovh/imgs/2022/05/22/8ea163c9c334972e.jpg';

describe('<Image />', () => {
  test('should render default', () => {
    const { container } = render(<Image src={src} />);
    expect(container).toMatchSnapshot();
  });

  test('should render image fallback', () => {
    const { container } = render(<Image src="undefined" fallback={src} />);
    expect(container).toMatchSnapshot();
  });
});
