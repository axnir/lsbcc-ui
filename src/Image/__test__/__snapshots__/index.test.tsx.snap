// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should render default 1`] = `
<div>
  <div
    class="lsbcc-image"
  >
    <div
      class="img-skeletion"
    />
  </div>
</div>
`;

exports[`<Image /> should render image fallback 1`] = `
<div>
  <div
    class="lsbcc-image"
  >
    <div
      class="img-skeletion"
    />
  </div>
</div>
`;
